test/unit/org/apache/cassandra/nodes/NodesTest.java

```diff
@@ -0,0 +1,347 @@
+import static org.mockito.Mockito.when;
+
+public class NodesTest
+{
+    private final ExecutorService executor = mock(ExecutorService.class);
+    private final INodesPersistence persistence = mock(INodesPersistence.class);
+    private final Future<?> promise = mock(Future.class);
+
+    private final UUID newHostId = UUID.randomUUID();
+    private final UUID id1 = UUID.randomUUID();
+    private final UUID id2 = UUID.randomUUID();
+    private final UUID id3 = UUID.randomUUID();
+
+    private Nodes nodes;
+    private static InetAddressAndPort addr1;
+    private static InetAddressAndPort addr2;
+    private static InetAddressAndPort addr3;
+    private ArgumentCaptor<Runnable> taskCaptor;
+    private AtomicReference<NodeInfo<?>> infoRef;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception
+    {
+        addr1 = InetAddressAndPort.getByNameOverrideDefaults("127.0.0.1", 7001);
+        addr2 = InetAddressAndPort.getByNameOverrideDefaults("127.0.0.2", 7001);
+        addr3 = InetAddressAndPort.getByNameOverrideDefaults("127.0.0.2", 7002);
+    }
+
+    @Before
+    public void beforeTest()
+    {
+        reset(persistence, executor, promise);
+        nodes = new Nodes(persistence, executor);
+        infoRef = new AtomicReference<>();
+        taskCaptor = ArgumentCaptor.forClass(Runnable.class);
+        when(executor.submit(taskCaptor.capture())).thenAnswer(inv -> promise);
+    }
+
+    @After
+    public void afterTest()
+    {
+        verify(persistence, atMostOnce()).loadLocal();
+        verify(persistence, atMostOnce()).loadPeers();
+        verifyNoMoreInteractions(executor, persistence, promise);
+    }
+
+    @Test
+    public void getEmptyLocal()
+    {
+        assertThat(nodes.getLocal().get()).isEqualTo(new LocalInfo());
+    }
+
+    @Test
+    public void getEmptyPeer()
+    {
+        assertThat(nodes.getPeers().get(addr1)).isNull();
+        assertThat(nodes.getPeers().get(addr2)).isNull();
+        assertThat(nodes.getPeers().get(addr3)).isNull();
+    }
+
+    @Test
+    public void loadLocal()
+    {
+        when(persistence.loadLocal()).thenReturn(new LocalInfo().setHostId(newHostId));
+        clearInvocations(persistence);
+        nodes = new Nodes(persistence, executor);
+
+        LocalInfo r = nodes.getLocal().get();
+        verify(persistence).loadLocal();
+        assertThat(r.getHostId()).isEqualTo(newHostId);
+    }
+
+    @Test
+    public void loadPeers()
+    {
+        List<PeerInfo> peers = Arrays.asList(new PeerInfo().setPeerAddressAndPort(addr1).setHostId(id1),
+                                             new PeerInfo().setPeerAddressAndPort(addr2).setHostId(id2),
+                                             new PeerInfo().setPeerAddressAndPort(addr3).setHostId(id3));
+        when(persistence.loadPeers()).thenReturn(peers.stream());
+        clearInvocations(persistence);
+        nodes = new Nodes(persistence, executor);
```

Comment:
We expect exact Nodes instance state for each test regardless of the tests execution order - that's why it is created for each test case.

I intentionally moved all mostly common initialization stuff out from the test case methods so that they are simpler and do not have unimportant details

Reply:
I see that it is created for each test in `beforeTest`. I am mainly wonder why it doesn't work here. Is it because of the two preceding lines (126 and 127), which are applied to `presistence` object?

Reply:
Ah... this is what you are asking for - Nodes object uses `load` methods during initialization. I want it to call mocked `INodesPersistence.loadPeers` because loading this is the subject of this test case